index_part.go

```diff
@@ -0,0 +1,157 @@
+/*
+ * go-leia
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package leia
+
+// IndexOption is the common type for indexing options
+// It should not be used directly
+type IndexOption interface{}
+
+// TransformerOption is the option for a FieldIndexer to apply transformation before indexing the value.
+// The transformation is also applied to a query value that matches the indexed field.
+type TransformerOption struct {
+	IndexOption
+	Transformer Transform
+}
+
+// TokenizerOption is the option for a FieldIndexer to split a value to be indexed into multiple parts.
+// Each part is then indexed separately.
+type TokenizerOption struct {
+	IndexOption
+	Tokenizer Tokenizer
+}
+
+// AliasOption is the option for a FieldIndexer to add a custom JSON path that will also resolve to the same Index part
+type AliasOption struct {
+	IndexOption
+	Alias string
+}
+
+// NewFieldIndexer creates a new fieldIndexer
+// leave the name empty to use the json path as name.
+// the name is to be used as query key when searching
+func NewFieldIndexer(jsonPath string, options ...IndexOption) FieldIndexer {
+	return fieldIndexer{
+		path:    jsonPath,
+		options: options,
+	}
+}
+
+type fieldIndexer struct {
+	path    string
+	options []IndexOption
+}
+
+func (j fieldIndexer) getAlias() *string {
+	for _, o := range j.options {
+		if a, b := o.(AliasOption); b {
+			return &a.Alias
+		}
+	}
+	return nil
+}
+
+func (j fieldIndexer) getTokenizer() Tokenizer {
+	for _, o := range j.options {
+		if a, b := o.(TokenizerOption); b {
+			return a.Tokenizer
+		}
+	}
+	return nil
+}
+
+func (j fieldIndexer) getTransformer() Transform {
```

Comment:
so you can't chain options of the same type (multiple transformers/tokenizers/aliases)?

Reply:
no